### PR TITLE
Add use_cudnn flag to BatchNormalization link

### PR DESCRIPTION
**Pull request by beam2d**
`batch_normalization` and `fixed_batch_normalization` functions support the `use_cudnn` argument, while `BatchNormalization` link does not have a corresponding attribute, which is added by this PR.

### CODE REVIEW

**Comment by gwtnb** [comment]
LGTM
